compiler/compile/Compilation.hpp

```diff
@@ -49,7 +49,8 @@ class OMR_EXTENSIBLE Compilation : public OMR::CompilationConnector
          TR::Options &options,
          TR::Region &heapMemoryRegion,
          TR_Memory *memory,
-         TR_OptimizationPlan *optimizationPlan) :
+         TR_OptimizationPlan *optimizationPlan,
+         bool shouldCompile = true) :
```

Comment:
Can you describe why we're introducing this parameter?

Reply:
I think this might be a question best asked and answered in #3019 which introduces it, even though it may be used in replay.

Reply:
shouldCompile was introduced along with "recordMethodBuilder(TR::MethodBuilder *m)" which is in "jitbuilder/control/Jit.cpp" This method makes use of "shouldCompile" so that the program being called only records (produce the .ilt file) and do not compile.

Reply:
So it seems the effort is duplicated in this PR? Why does both #3056 and #3019 have the exact same change?